server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java

```diff
@@ -5827,4 +5830,28 @@ public void mimeMessageIdShouldBePreservedWhenSending() {
         assertThat(receivedMimeMessageId).isEqualTo(creationMimeMessageId);
     }
 
+    @Test
+    public void setMessagesShouldWorkForHugeNumberOfEmailsToTrash() throws Exception {
+        ZonedDateTime dateTime = ZonedDateTime.parse("2014-10-30T14:12:00Z");
+        String mailIds = IntStream
+            .rangeClosed(1, NUMBER_OF_MAIL_TO_CREATE)
+            .mapToObj(Throwing.intFunction((i) ->
+                mailboxProbe
+                    .appendMessage(USERNAME, MailboxPath.forUser(USERNAME, DefaultMailboxes.TRASH),
+                        new ByteArrayInputStream("Subject: my test subject\r\n\r\ntestmail".getBytes(StandardCharsets.UTF_8)), Date.from(dateTime.toInstant()), false, new Flags())
+                    .getMessageId()
+                    .serialize()).sneakyThrow())
+            .collect(Collectors.joining("\", \""));
```

Comment:
I'd prefer:

```
.map(this::quote)
.collect(Collectors.joining(","))
```

That would be easier to read and the intent would be better expressed.

Reply:
This method doesn't exist.